index/index.go

```diff
@@ -781,9 +815,10 @@ func ReadOffsetTable(bs ByteSlice, off uint64, f func([]string, uint64) error) e
 	d := encoding.NewDecbufAt(bs, int(off), castagnoliTable)
 	cnt := d.Be32()
 
+	keys := make([]string, 0, 2)
```

Comment:
any particular reason for 2?

Reply:
As per the current use case, not more than 2 keys are required
https://github.com/prometheus/tsdb/blob/c7af8aa9f3752aa39a6602f8143e44ee41485ecd/index/index.go#L717-L740

Reply:
should we be checking the value of `d.Uvarint` to make sure it's 2 then? because we append `keyCount` worth of keys to `keys`

Reply:
As per the current design, the keyCount doesn't exceed 2.

Postings table has keyCount of 2 (and I think it will remain 2).
https://github.com/prometheus/tsdb/blob/c7af8aa9f3752aa39a6602f8143e44ee41485ecd/index/index.go#L471

And Label Index table has keyCount of 1
https://github.com/prometheus/tsdb/blob/c7af8aa9f3752aa39a6602f8143e44ee41485ecd/index/index.go#L451
https://github.com/prometheus/tsdb/blob/c7af8aa9f3752aa39a6602f8143e44ee41485ecd/compact.go#L852

During my GSoC last year, there was a discussion about having label indices for composite label names, but it was decided that we don't need it though the design still allows it. So I don't expect it to exceed 1 anytime soon.

Reply:
I just feel like it's worth codifying (or at least leaving a comment) somewhere about this. It wasn't even clear to me what the keys were initially.